magefile.go

```diff
@@ -15,15 +15,147 @@ import (
 	"path/filepath"
 	"runtime"
 	"strings"
+	"time"
 
+	goreleaserConfig "github.com/goreleaser/goreleaser/pkg/config"
 	"github.com/magefile/mage/mg"
 	"github.com/magefile/mage/sh"
 	"github.com/pkg/errors"
+	"golang.org/x/exp/maps"
+	"sigs.k8s.io/yaml"
+
+	"github.com/G-Research/armada/pkg/client/util"
 )
 
 const PROTOC_VERSION_MIN = "3.21.8"
 const PROTOC_VERSION_DOWNLOAD = "21.8" // The "3." is omitted.
 
+// Build images, spin up a test environment, and run the integration tests against it.
+func CiIntegrationTests() error {
+	if err := os.MkdirAll(".kube", os.ModeDir); err != nil {
+		return err
+	}
+	mg.Deps(BootstrapTools)
+	mg.Deps(DockerBundle, Kind)
+	err := sh.Run("docker-compose", "up", "-d", "redis", "postgres", "pulsar", "stan")
+	if err != nil {
+		return err
+	}
+
+	// TODO: Necessary to avoid connection error on Armada server startup.
+	time.Sleep(10 * time.Second)
```

Comment:
are we able to check some condition, to make this less likely to fail in the case of slow startup?

Reply:
I think we should fix Armada to retry for some time.